### ISSUE TITLE
timeout does not work with AddRabbitMQ invalid endpoint 

### ISSUE THREAD

**Issue by robsonrjan** [Xabaril/AspNetCore.Diagnostics.HealthChecks#714]
Please, fill the following sections to help us fix the issue

**What happened**:
When rabbitmq endpoint address is nor valid the health check taks 40s against the given timeout 10s.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
builder.AddRabbitMQ(
                rabbitConnectionString: $"amqps://{UserName}:{Password}@{InvalidHostName}:{Port}",
                sslOption: default,
                name: "RabbitMQ",
                failureStatus: HealthStatus.Unhealthy,
                tags: new[] { tagName },
                timeout: TimeSpan.FromSeconds(10));
                
**Source code sample**:

using var cancellationToken = new CancellationTokenSource(TimeSpan.FromSeconds(10));
var healthCheckTask = healthCheckService.CheckHealthAsync(cancellationToken.Token);
Takes up to 40s also.

**Anything else we need to know?**:

**Environment**:
kubernetes - linux

- .NET Core version
netcoreapp3.1
- Healthchecks version
HealthChecks.RabbitMQ 3.1.4
- Operative system:
linux
- Others:

**Comment by sungam3r** [comment]
Indeed. `CancellationToken` is not used by `RabbitMQHealthCheck.CheckHealthAsync` method. `ConnectionFactory` accepts `RequestedConnectionTimeout` setting but it is just plain timeout, not cancellation token. At the end this looks like a limitation of the underlying core APIs (`Socket.ConnectAsync`), used by rabbit-dotnet-client, see https://github.com/dotnet/runtime/issues/19592. #714 helps to mitigate issue. I would like to keep this issue open for now.
